core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -215,6 +215,7 @@
     private final AtomicLong requestIdGenerator = new AtomicLong();
     private final CounterMetric numHandshakes = new CounterMetric();
     private static final String HANDSHAKE_ACTION_NAME = "internal:tcp/handshake";
+    private volatile boolean closingTransport = false;
```

Comment:
I think we can just use `this.stopped()` instead of having a separate boolean? we are setting the state value before we execute the `doStop` method

Reply:
Great suggestion. I made that change.